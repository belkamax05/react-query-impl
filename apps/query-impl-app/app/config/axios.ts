import axios from 'axios';

const Axios = axios.create({});

export default Axios;
