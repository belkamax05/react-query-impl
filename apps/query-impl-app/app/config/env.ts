export const START_TIME = new Date();
export const ENV_DEV = process.env.NODE_ENV !== 'production';

/**
 * Check if the code is running on the client (browser have window object)
 */
export const IS_CLIENT = typeof window !== 'undefined';
