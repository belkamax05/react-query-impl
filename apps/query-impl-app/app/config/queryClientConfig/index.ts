import { QueryClientConfig } from '@tanstack/react-query';

/**
 * QueryClient's instance config for app
 */
const queryClientConfig: QueryClientConfig = {
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
      retry: false,
      staleTime: Infinity,
      cacheTime: Infinity,
    },
  },
};

export default queryClientConfig;
