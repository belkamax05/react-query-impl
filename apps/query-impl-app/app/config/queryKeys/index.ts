import {
  DeepRecord,
  QueryKeyFunction,
  validateConfigType,
} from '@query-impl/core';
import { QueryKey } from '@tanstack/react-query';

const queryKeys = validateConfigType(
  {
    app: {
      dashboard: ['app', 'dashboard'],
    },
    samples: {
      counter: ['samples', 'counter'],
      todo: ['samples', 'todo'],
    },
  } as const,
  null as DeepRecord<string, QueryKey | QueryKeyFunction>
);

export default queryKeys;
