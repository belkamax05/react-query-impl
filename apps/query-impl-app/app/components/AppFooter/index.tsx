import { START_TIME } from '../../config/env';
import styles from './styles.module.scss';

const AppFooter = () => {
  return (
    <footer className={styles['root']}>
      <pre>Startup time: {START_TIME.toISOString()}</pre>
      <pre>Init time: {new Date().toISOString()}</pre>
    </footer>
  );
};

export default AppFooter;
