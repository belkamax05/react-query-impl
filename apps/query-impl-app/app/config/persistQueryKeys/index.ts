import { QueryKeyJoin } from '@query-impl/core';
import { Call, Tuples } from 'hotscript';
import queryKeys from '../queryKeys';
/**
 * list of query keys to be saved into localStorage and restored on each session
 */
const persistQueryKeys = [queryKeys.samples.counter, queryKeys.samples.todo];

export type PersistQueryKeys = typeof persistQueryKeys;

export type PersistQueryKeysUnion = Call<Tuples.ToUnion, PersistQueryKeys>;

export type PersistQueryKeysKey = QueryKeyJoin<PersistQueryKeysUnion>;

export default persistQueryKeys;
