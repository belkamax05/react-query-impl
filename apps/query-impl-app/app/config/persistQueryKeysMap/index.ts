import persistQueryKeys, { PersistQueryKeysKey } from '../persistQueryKeys';

const persistQueryKeysMap = (<
  TKey extends PersistQueryKeysKey,
  TResult extends Record<TKey, true>
>(): TResult => {
  const result: Partial<TResult> = {};
  persistQueryKeys.forEach((queryKey) => {
    const jointKey = queryKey.join('.') as TKey;
    result[jointKey] = true as TResult[TKey];
  });
  return result as TResult;
})();

export default persistQueryKeysMap;
