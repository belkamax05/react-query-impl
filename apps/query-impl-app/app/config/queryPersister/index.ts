import { createSyncStoragePersister } from '@tanstack/query-sync-storage-persister';
import { PersistedClient } from '@tanstack/react-query-persist-client';
import { IS_CLIENT } from '../env';
import persistQueryKeysMap from '../persistQueryKeysMap';

export const persisterSerialize = (client: PersistedClient) => {
  const resultClient: PersistedClient = {
    ...client,
    clientState: {
      ...client.clientState,
      queries: client.clientState.queries.filter((query) => {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const res = (persistQueryKeysMap as any)[query.queryKey.join('.') as any];
        return res;
      }),
    },
  };
  return JSON.stringify(resultClient);
};

export const persisterDeserialize = (cachedString: string) => {
  const client: PersistedClient = JSON.parse(cachedString);
  return client;
};

const queryPersister = createSyncStoragePersister({
  key: 'persist-query',
  storage: IS_CLIENT ? window.localStorage : null,
  serialize: persisterSerialize,
  deserialize: persisterDeserialize,
});

export default queryPersister;
