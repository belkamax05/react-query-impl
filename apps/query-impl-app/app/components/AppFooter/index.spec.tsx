import { render } from '@testing-library/react';
import AppFooter from '.';
import { START_TIME } from '../../config/env';

describe('AppFooter', () => {
  it('should render the footer with the correct startup and init times', async () => {
    const { baseElement, getByText } = render(<AppFooter />);
    const startupTime = getByText(`Startup time: ${START_TIME.toISOString()}`);
    const initTime = getByText(`Init time: ${new Date().toISOString()}`);
    expect(startupTime).toBeInTheDocument();
    expect(initTime).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
