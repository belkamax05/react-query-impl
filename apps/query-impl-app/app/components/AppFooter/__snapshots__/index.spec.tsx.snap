// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter should render the footer with the correct startup and init times 1`] = `
<body>
  <div>
    <footer
      class="root"
    >
      <pre>
        Startup time: 
        2020-01-01T00:00:00.000Z
      </pre>
      <pre>
        Init time: 
        2020-01-01T00:00:00.000Z
      </pre>
    </footer>
  </div>
</body>
`;
